### ISSUE TITLE
Not compatible with Chrome and Internet Explorer

### ISSUE THREAD

**Issue by kamilszkoda** [Landoop/kafka-topics-ui#40]
The newest version doesn't work properly with Chrome and Internet Explorer. Only Microsoft Edge show the whole website properly. For example at Chrome occurs connection error which doesn't appear at Edge.

**Comment by chdask** [comment]
Hi @kamilszkoda, could you give us more details for the Chrome error you are getting (like screenshot, console output..) ? 
Thanks for raising.

**Comment by kamilszkoda** [comment]
Yes. I attach screens from 3 browsers. Only on Edge it works properly

![chrome](https://cloud.githubusercontent.com/assets/25080343/21893669/545ba94a-d8dc-11e6-8007-2a17fc84af39.png)
![internet_explorer](https://cloud.githubusercontent.com/assets/25080343/21893671/55c5437c-d8dc-11e6-9976-4218b464b239.png)
![microsoft_edge](https://cloud.githubusercontent.com/assets/25080343/21893673/575d815e-d8dc-11e6-8f20-16f11d098e9b.png)

**Comment by kamilszkoda** [comment]
1.  Chrome
2. Internet Explorer
3. Microsoft Edge

On each browser I opened the same link. I have the same network and proxy configuration.

**Comment by chdask** [comment]
Interesting, I am using chrome but works fine. 
Do you have any error on the console? 
I don't think IE is supported, maybe we can incorporate this in the docs, but what is the version you are using?

**Comment by kamilszkoda** [comment]
Hi,

I can't find any errors. I use the newest version:
Info about version and run:

REPOSITORY                               TAG                 IMAGE ID            CREATED             SIZE
landoop/schema-registry-ui      latest              7475d58eea4b        6 days ago          24.8 MB
landoop/kafka-topics-ui            latest              56c8f8981e1a        6 days ago          25.21 MB

CONTAINER ID        IMAGE                        COMMAND             CREATED             STATUS              PORTS                    NAMES
3439fc715954        landoop/kafka-topics-ui      "/run.sh"           3 days ago          Up 3 days           0.0.0.0:8001->8000/tcp   hungry_booth
aaf93d1d8b8a        landoop/schema-registry-ui   "/run.sh"           3 days ago          Up 3 days           0.0.0.0:8000->8000/tcp   nostalgic_agnesi

My running scripts are:
docker run --rm -i \
-p 8000:8000 \
-e SCHEMAREGISTRY_URL=http://kafka-dev.dw.gc.stepstone.com:8081 landoop/schema-registry-ui

docker run --rm -i \
--net=host \
-p 8001:8000 \
-e KAFKA_REST_PROXY_URL=http://kafka-dev.dw.gc.stepstone.com:8082 landoop/kafka-topics-ui

**Comment by kamilszkoda** [comment]
Url : http://kafka-dev.dw.gc.stepstone.com:8081 
Global Compatibility level : BACKWARD
schema-registry-ui: 0.8.0

Kafka Rest : http://kafka-dev.dw.gc.stepstone.com:8082 
Kafka Brokers : 1
kafka-topics-ui: 0.8.1

**Comment by smifun** [comment]
I run into similar issue. I got following in chrome console:

```
XMLHttpRequest cannot load http://localhost:8082/topics. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:8000' is therefore not allowed access.
angular.js:14110 

Error in getting topics from kafka-rest : {"data":null,"status":-1,"config":{"method":"GET","transformRequest":[null],"transformResponse":[null],"url":"http://localhost:8082/topics","headers":{"Accept":"application/json, text/plain, */*"}},"statusText":""}
(anonymous) @ angular.js:14110

angular.js:14110 Failed: Error in getting topics from kafka-rest
(anonymous) @ angular.js:14110
:8000/#/cluster/default:1 

XMLHttpRequest cannot load http://localhost:8082/consumers/Consumer-1486055359835-json. Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:8000' is therefore not allowed access.

angular.js:14110 TypeError: Cannot read property 'message' of null
    at failure (combined.js?rel=cb6ae59acb:1114)
    at angular.js:16606
    at m.$eval (angular.js:17913)
    at m.$digest (angular.js:17727)
    at m.$apply (angular.js:18021)
    at l (angular.js:12002)
    at D (angular.js:12211)
    at XMLHttpRequest.e (angular.js:12149)
(anonymous) @ angular.js:14110
```

here are some logs from kafka-rest

```
[2017-02-02 18:09:19,844] INFO 0:0:0:0:0:0:0:1 - - [02/Feb/2017:18:09:19 +0100] "GET /brokers HTTP/1.1" 200 15  8 (io.confluent.rest-utils.requests:77)
[2017-02-02 18:09:19,845] INFO 0:0:0:0:0:0:0:1 - - [02/Feb/2017:18:09:19 +0100] "GET /topics HTTP/1.1" 200 158  3 (io.confluent.rest-utils.requests:77)
[2017-02-02 18:09:19,853] INFO 0:0:0:0:0:0:0:1 - - [02/Feb/2017:18:09:19 +0100] "OPTIONS /consumers/Consumer-1486055359835-json HTTP/1.1" 200 1201  12 (io.confluent.rest-utils.requests:77)
```

I tried to open it on Safari but with no luck.


```
Kafka Rest : http://localhost:8082 
Kafka Brokers : 
kafka-topics-ui: 0.8.1 CONNECTIVITY ERROR
```

**Comment by Antwnis** [comment]
Hi @smifun 

The issue that you are describing is different to the IE / old Chrome issue
The security of browsers identifies that requests are coming in from different locations:
i. The micro-service schema-registry
ii. The web server serving the UI (schema-registry-ui)

This is called CORS (Cross-Origin Resource Sharing)

To overcome it - if you use nginx web server check https://github.com/Landoop/kafka-topics-ui#nginx-config - or alternatively you can use the docker solution to get going as it contains the necessary CORS configuration

**Comment by ktlim86** [comment]
Hi,

I have the same issue. but this time round it is another problem.
Here is the error message from the chrome console.

```
x.x.4.155:8082 Starting kafka-topics controller : config
angular.js:14110   curl -X GET x.x.4.155:8082/brokers
angular.js:14110 Starting kafka-topics controller : home
angular.js:14110 Error: Failed to execute 'open' on 'XMLHttpRequest': Invalid URL
    at angular.js:12115
    at n (angular.js:11947)
    at angular.js:11742
    at angular.js:16606
    at m.$eval (angular.js:17913)
    at m.$digest (angular.js:17727)
    at m.$apply (angular.js:18021)
    at l (angular.js:12002)
    at D (angular.js:12211)
    at XMLHttpRequest.w.onload (angular.js:12139)
(anonymous) @ angular.js:14110
(anonymous) @ angular.js:10633
(anonymous) @ angular.js:16614
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
l @ angular.js:12002
D @ angular.js:12211
w.onload @ angular.js:12139
angular.js:14110 Failure with : {"stack":"Error: Failed to execute 'open' on 'XMLHttpRequest': Invalid URL\n    at http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:103:351\n    at n (http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:100:147)\n    at http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:97:364\n    at http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:132:20\n    at m.$eval (http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:146:293)\n    at m.$digest (http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:143:420)\n    at m.$apply (http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:147:78)\n    at l (http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:98:421)\n    at D (http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:103:105)\n    at XMLHttpRequest.w.onload (http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:104:77)"}
(anonymous) @ angular.js:14110
failure @ kafka-rest-proxy-factory.js:404
(anonymous) @ angular.js:16606
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
l @ angular.js:12002
D @ angular.js:12211
w.onload @ angular.js:12139
angular.js:14110 Creating Kafka Rest consumer for json data
angular.js:14110   curl -X POST -H "Content-Type: application/vnd.kafka.v1+json" --data '{"name": "instance", "format": "json", "auto.offset.reset": "smallest"}' x.x.4.155:8082/consumers/Consumer-1490836967746-json
angular.js:14110   curl -X GET x.x.4.155:8082/topics
2angular.js:14110 Error: Failed to execute 'open' on 'XMLHttpRequest': Invalid URL
    at angular.js:12115
    at n (angular.js:11947)
    at angular.js:11742
    at angular.js:16606
    at m.$eval (angular.js:17913)
    at m.$digest (angular.js:17727)
    at m.$apply (angular.js:18021)
    at l (angular.js:12002)
    at D (angular.js:12211)
    at XMLHttpRequest.w.onload (angular.js:12139)
(anonymous) @ angular.js:14110
(anonymous) @ angular.js:10633
(anonymous) @ angular.js:16614
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
l @ angular.js:12002
D @ angular.js:12211
w.onload @ angular.js:12139
angular.js:14110 TypeError: Cannot read property 'message' of undefined
    at failure (kafka-rest-proxy-factory.js:226)
    at angular.js:16606
    at m.$eval (angular.js:17913)
    at m.$digest (angular.js:17727)
    at m.$apply (angular.js:18021)
    at l (angular.js:12002)
    at D (angular.js:12211)
    at XMLHttpRequest.w.onload (angular.js:12139)
(anonymous) @ angular.js:14110
(anonymous) @ angular.js:10633
(anonymous) @ angular.js:16614
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
l @ angular.js:12002
D @ angular.js:12211
w.onload @ angular.js:12139
angular.js:14110 Error in getting topics from kafka-rest : {"stack":"Error: Failed to execute 'open' on 'XMLHttpRequest': Invalid URL\n    at http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:103:351\n    at n (http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:100:147)\n    at http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:97:364\n    at http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:132:20\n    at m.$eval (http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:146:293)\n    at m.$digest (http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:143:420)\n    at m.$apply (http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:147:78)\n    at l (http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:98:421)\n    at D (http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:103:105)\n    at XMLHttpRequest.w.onload (http://x.x.4.155/kafka-ui/bower_components/angular/angular.min.js:104:77)"}
(anonymous) @ angular.js:14110
failure @ kafka-rest-proxy-factory.js:31
(anonymous) @ angular.js:16606
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
l @ angular.js:12002
D @ angular.js:12211
w.onload @ angular.js:12139
angular.js:14110 Failed: Error in getting topics from kafka-rest
```

just to let you know i deploy this from the github and I have a nginx server deployed as described [here](https://github.com/Landoop/kafka-topics-ui#nginx-config).  However, after deploying, I still have this problem. Take note I have both Kafka, Kafka REST and the kafka-topics-ui deployed in the same machine. Please advise. Thanks. :)

**Comment by Antwnis** [comment]
@ktlim86 Your issue seems to be
    
    'XMLHttpRequest': Invalid URL

Try adding the protocol http:// in when you define the Rest Proxy address, i.e.

    http://x.x.4.155:8082

And let us know if that fixes the issue for you.

**Comment by ktlim86** [comment]
@Antwnis manage to fix the issue. However, I am back to the same situation as [this](https://github.com/Landoop/kafka-topics-ui/issues/40#issuecomment-272177701).

On Edge browser I can see. But in chrome, it complains that connection error and the following is the error from chrome console. Please advise. Thanks :)

```
ARIA: Attribute " aria-label ", required for accessibility, is missing on node: <md-select ng-model=​"connectEndPoint" class=​"ng-pristine ng-untouched ng-valid ng-empty ng-not-empty-add ng-pristine-remove ng-dirty-add ng-valid-parse-add ng-animate ng-empty-remove" tabindex=​"0" aria-disabled=​"false" role=​"listbox" aria-expanded=​"false" aria-multiselectable=​"false" id=​"select_1" aria-owns=​"select_container_2" aria-invalid=​"false" data-ng-animate=​"2">​…​</md-select>​
(anonymous) @ angular.js:14110
r @ angular-material.min.js:7
(anonymous) @ angular-material.min.js:11
(anonymous) @ angular.js:1259
la @ angular.js:10095
p @ angular.js:9492
g @ angular.js:8757
p @ angular.js:9486
g @ angular.js:8757
p @ angular.js:9486
g @ angular.js:8757
p @ angular.js:9486
g @ angular.js:8757
p @ angular.js:9486
g @ angular.js:8757
(anonymous) @ angular.js:8637
(anonymous) @ angular.js:1808
$eval @ angular.js:17913
$apply @ angular.js:18013
(anonymous) @ angular.js:1806
invoke @ angular.js:4762
c @ angular.js:1804
Ic @ angular.js:1824
ue @ angular.js:1709
(anonymous) @ angular.js:32379
b @ angular.js:3251
gg @ angular.js:3540
d @ angular.js:3528
angular.js:14110 Starting kafka-topics controller : list (getting topic info)
angular.js:14110 ARIA: Attribute " aria-label ", required for accessibility, is missing on node: <md-select ng-model=​"connectEndPoint" class=​"ng-pristine ng-untouched ng-valid ng-empty ng-not-empty-add ng-pristine-remove ng-dirty-add ng-valid-parse-add ng-animate ng-empty-remove" tabindex=​"0" aria-disabled=​"false" role=​"listbox" aria-expanded=​"false" aria-multiselectable=​"false" id=​"select_1" aria-owns=​"select_container_2" aria-invalid=​"false" data-ng-animate=​"2">​…​</md-select>​
(anonymous) @ angular.js:14110
r @ angular-material.min.js:7
m @ angular-material.min.js:11
(anonymous) @ angular-material.min.js:11
$digest @ angular.js:17806
$apply @ angular.js:18021
(anonymous) @ angular.js:1806
invoke @ angular.js:4762
c @ angular.js:1804
Ic @ angular.js:1824
ue @ angular.js:1709
(anonymous) @ angular.js:32379
b @ angular.js:3251
gg @ angular.js:3540
d @ angular.js:3528
angular.js:12185 XHR finished loading: GET "http://x.x.4.155/kafka-ui/src/kafka-topics/config/config.html".
(anonymous) @ angular.js:12185
n @ angular.js:11947
(anonymous) @ angular.js:11742
(anonymous) @ angular.js:16606
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
(anonymous) @ angular.js:1806
invoke @ angular.js:4762
c @ angular.js:1804
Ic @ angular.js:1824
ue @ angular.js:1709
(anonymous) @ angular.js:32379
b @ angular.js:3251
gg @ angular.js:3540
d @ angular.js:3528
angular.js:12185 XHR finished loading: GET "http://x.x.4.155/kafka-ui/src/kafka-topics/list/left-list.html".
(anonymous) @ angular.js:12185
n @ angular.js:11947
(anonymous) @ angular.js:11742
(anonymous) @ angular.js:16606
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
(anonymous) @ angular.js:1806
invoke @ angular.js:4762
c @ angular.js:1804
Ic @ angular.js:1824
ue @ angular.js:1709
(anonymous) @ angular.js:32379
b @ angular.js:3251
gg @ angular.js:3540
d @ angular.js:3528
angular.js:12185 XHR finished loading: GET "http://x.x.4.155/kafka-ui/src/kafka-topics/home/home.html".
(anonymous) @ angular.js:12185
n @ angular.js:11947
(anonymous) @ angular.js:11742
(anonymous) @ angular.js:16606
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
(anonymous) @ angular.js:1806
invoke @ angular.js:4762
c @ angular.js:1804
Ic @ angular.js:1824
ue @ angular.js:1709
(anonymous) @ angular.js:32379
b @ angular.js:3251
gg @ angular.js:3540
d @ angular.js:3528
angular.js:14110 http://x.x.4.155:8082 Starting kafka-topics controller : config
angular.js:14110   curl -X GET http://x.x.4.155:8082/brokers
angular.js:14110 Starting kafka-topics controller : home
angular.js:14110 Creating Kafka Rest consumer for json data
angular.js:14110   curl -X POST -H "Content-Type: application/vnd.kafka.v1+json" --data '{"name": "instance", "format": "json", "auto.offset.reset": "smallest"}' http://x.x.4.155:8082/consumers/Consumer-1490839943150-json
angular.js:14110   curl -X GET http://x.x.4.155:8082/topics
angular.js:12185 XHR finished loading: GET "http://x.x.4.155/kafka-ui/src/kafka-topics/list/topics-list.html".
(anonymous) @ angular.js:12185
n @ angular.js:11947
(anonymous) @ angular.js:11742
(anonymous) @ angular.js:16606
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
l @ angular.js:12002
D @ angular.js:12211
w.onload @ angular.js:12139
angular.js:12185 XHR finished loading: GET "http://x.x.4.155/kafka-ui/src/kafka-topics/list/control-topics-list.html".
(anonymous) @ angular.js:12185
n @ angular.js:11947
(anonymous) @ angular.js:11742
(anonymous) @ angular.js:16606
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
l @ angular.js:12002
D @ angular.js:12211
w.onload @ angular.js:12139
index.html#/:1 XMLHttpRequest cannot load http://x.x.4.155:8082/brokers. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://x.x.4.155' is therefore not allowed access.
angular.js:12185 XHR failed loading: GET "http://x.x.4.155:8082/brokers".
(anonymous) @ angular.js:12185
n @ angular.js:11947
(anonymous) @ angular.js:11742
(anonymous) @ angular.js:16606
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
l @ angular.js:12002
D @ angular.js:12211
w.onload @ angular.js:12139
angular.js:14110 Failure with : {"data":null,"status":-1,"config":{"method":"GET","transformRequest":[null],"transformResponse":[null],"url":"http://x.x.4.155:8082/brokers","headers":{"Accept":"application/json, text/plain, */*"}},"statusText":""}
(anonymous) @ angular.js:14110
failure @ kafka-rest-proxy-factory.js:404
(anonymous) @ angular.js:16606
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
l @ angular.js:12002
D @ angular.js:12211
e @ angular.js:12149
index.html#/:1 XMLHttpRequest cannot load http://x.x.4.155:8082/topics. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://x.x.4.155' is therefore not allowed access.
angular.js:12185 XHR failed loading: GET "http://x.x.4.155:8082/topics".
(anonymous) @ angular.js:12185
n @ angular.js:11947
(anonymous) @ angular.js:11742
(anonymous) @ angular.js:16606
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
l @ angular.js:12002
D @ angular.js:12211
w.onload @ angular.js:12139
angular.js:14110 Error in getting topics from kafka-rest : {"data":null,"status":-1,"config":{"method":"GET","transformRequest":[null],"transformResponse":[null],"url":"http://x.x.4.155:8082/topics","headers":{"Accept":"application/json, text/plain, */*"}},"statusText":""}
(anonymous) @ angular.js:14110
failure @ kafka-rest-proxy-factory.js:31
(anonymous) @ angular.js:16606
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
l @ angular.js:12002
D @ angular.js:12211
e @ angular.js:12149
angular.js:14110 Failed: Error in getting topics from kafka-rest
(anonymous) @ angular.js:14110
(anonymous) @ topics-list.controller.js:117
(anonymous) @ angular.js:16606
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
l @ angular.js:12002
D @ angular.js:12211
e @ angular.js:12149
index.html#/:1 XMLHttpRequest cannot load http://192.168.4.155:8082/consumers/Consumer-1490839943150-json. Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://192.168.4.155' is therefore not allowed access.
angular.js:12185 XHR failed loading: POST "http://192.168.4.155:8082/consumers/Consumer-1490839943150-json".
(anonymous) @ angular.js:12185
n @ angular.js:11947
(anonymous) @ angular.js:11742
(anonymous) @ angular.js:16606
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
l @ angular.js:12002
D @ angular.js:12211
w.onload @ angular.js:12139
angular.js:14110 TypeError: Cannot read property 'message' of null
    at failure (kafka-rest-proxy-factory.js:226)
    at angular.js:16606
    at m.$eval (angular.js:17913)
    at m.$digest (angular.js:17727)
    at m.$apply (angular.js:18021)
    at l (angular.js:12002)
    at D (angular.js:12211)
    at XMLHttpRequest.e (angular.js:12149)
(anonymous) @ angular.js:14110
(anonymous) @ angular.js:10633
(anonymous) @ angular.js:16614
$eval @ angular.js:17913
$digest @ angular.js:17727
$apply @ angular.js:18021
l @ angular.js:12002
D @ angular.js:12211
e @ angular.js:12149
```

this is my env.js
```
var clusters = [{
        NAME: "prod",
        KAFKA_REST: "http://x.x.4.155:8082",
        MAX_BYTES: "?max_bytes=50000",  // Sets the default maximum amount of bytes to fetch from each topic
        COLOR: "#141414" // Optional
    }];
```

this is my nginx config.
```
server {

        root /var/www/html/kafka-ui;

        index index.html index.htm index.nginx-debian.html;
        server_name _;
        location /kafka-ui {
                add_header 'Access-Control-Allow-Origin' "$http_origin" always;
                add_header 'Access-Control-Allow-Credentials' 'true' always;
                add_header 'Access-Control-Allow-Methods' 'GET, POST, PUT, DELETE, OPTIONS' always;
                add_header 'Access-Control-Allow-Headers' 'Accept,Authorization,Cache-Control,Content-Type,DNT,If-Modified-Since,Keep-Alive,Origin,User-Agent,X-Mx-ReqToken,X-Requested-With' always;

                proxy_pass http://x.x.4.155:8082;
                proxy_redirect off;

                proxy_set_header  X-Real-IP  $remote_addr;
                proxy_set_header  X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header  Host $http_host;
        }
}
```

**Comment by brusser** [comment]
I am also seeing the connectivity error issue on Chrome. I even see it on the http://kafka-topics-ui.landoop.com/#/ demo site. Don't see a solution above and enabling CORS does not seem to fix it. Is this a problem that is specific to me? 

I should mention I am using the Enterprise version of Chrome - Version 61.0.3163.100 (Official Build) (64-bit)

**Comment by brusser** [comment]
For me I found the issue to be related to running on localhost and a bug in Chrome for CORS. Error I was getting in the Chrome console was: Failed to load http://localhost:8082/topics: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:8091' is therefore not allowed access.

localhost:8082 is is the Kafka rest port and localhost:8092 is the kafka-topics-ui port.

Found a workaround here: https://bugs.chromium.org/p/chromium/issues/detail?id=67743#c34

Run Chrome with the argument "--disable-web-security" and with a new user profile

**UPDATE - SOLUTION:**
 Proper fix is to setup kafka-rest properly. Add

access.control.allow.origin=*
access.control.allow.methods=GET,POST,PUT,OPTIONS

to the kafka-rest.properties.

Found it here: https://stackoverflow.com/questions/37561271/kafka-connect-rest-api-cross-domain-exception-invoking-from-ajax#37581873

Therefore this is not an issue with kafka-topics-ui. At least not for me.
